### PR TITLE
Move login to redux

### PR DESCRIPTION
**Pull request by neilff**
I added a promise middleware for resolving the response from the mock API. Show / hiding the error is now based on the app state and not local component state.

I'm not sure if this is taking the seed too far though..

### CODE REVIEW

**Comment by neilff** [comment]
@andrejkn @rbrander can you take a look at these potential changes and provide feedback?

File: src/utils/utilities.js [hunk 1]
```diff
@@ -16,3 +16,15 @@ export function immutableToJS(state) {
...
 }
+
+/**
+ * Returns whether the provided value is a promise
+ *
+ * @param {object} value Potential promise
+ * @return {Boolean}
+ */
+export function isPromise(value) {
+  if (value !== null && typeof value === 'object') {
```
Comment (rbrander) [inline src/utils/utilities.js]: it might be a good idea to add a unit test for this

**Comment by rbrander** [comment]
The code looks great overall.  I'd recommend you squash your commits before merging.  Other than that, LGTM. :shipit:

**Comment by neilff** [comment]
@rbrander, @andrejkn 
- Added unit tests
- Added timeout for login mock, simulate a delayed api

Still need to do some more work, but will submit as additional PRs after this is merged

**Comment by andrejkn** [comment]
@neilff looks good :shipit:

**Comment by neilff** [comment]
I actually don't have the ability to merge, feel free to do so once you are cool with it :+1:

**Comment by andrejkn** [comment]
@rbrander we need WRITE permission :)

**Pull request merged**